src/compiler-debugging.md

```diff
@@ -0,0 +1,320 @@
+---
+layout: default
+title: Debugging the Compiler
+---
+
+# Debugging the compiler
+[debugging]: #debugging
+
+Here are a few tips to debug the compiler:
+
+## Getting a backtrace
+[getting-a-backtrace]: #getting-a-backtrace
+
+When you have an ICE (panic in the compiler), you can set
+`RUST_BACKTRACE=1` to get the stack trace of the `panic!` like in
+normal Rust programs.  IIRC backtraces **don't work** on Mac and on MinGW,
+sorry. If you have trouble or the backtraces are full of `unknown`,
+you might want to find some way to use Linux or MSVC on Windows.
+
+In the default configuration, you don't have line numbers enabled, so the 
+backtrace looks like this:
+
+   ```
+stack backtrace:
+   0: std::sys::imp::backtrace::tracing::imp::unwind_backtrace
+   1: std::sys_common::backtrace::_print
+   2: std::panicking::default_hook::{{closure}}
+   3: std::panicking::default_hook
+   4: std::panicking::rust_panic_with_hook
+   5: std::panicking::begin_panic
+   6: rustc_typeck::check::cast::<impl rustc_typeck::check::FnCtxt<'a, 'gcx, \ 
+      'tcx>>::pointer_kind
+   (~~~~ LINES REMOVED BY ME FOR BREVITY ~~~~)
+  32: rustc_typeck::check_crate
+  33: <std::thread::local::LocalKey<T>>::with
+  34: <std::thread::local::LocalKey<T>>::with
+  35: rustc::ty::context::TyCtxt::create_and_enter
+  36: rustc_driver::driver::compile_input
+  37: rustc_driver::run_compiler
+   ```
+
+If you want line numbers for the stack trace, you can enable 
+`debuginfo-lines=true` or `debuginfo=true` in your config.toml and rebuild the 
+compiler. Then the backtrace will look like this:
+
+```
+stack backtrace:
+   (~~~~ LINES REMOVED BY ME FOR BREVITY ~~~~)
+   6: rustc_typeck::check::cast::<impl rustc_typeck::check::FnCtxt<'a, 'gcx, \
+             'tcx>>::pointer_kind
+             at /home/user/rust/src/librustc_typeck/check/cast.rs:110
+   7: rustc_typeck::check::cast::CastCheck::check
+             at /home/user/rust/src/librustc_typeck/check/cast.rs:572
+             at /home/user/rust/src/librustc_typeck/check/cast.rs:460
+             at /home/user/rust/src/librustc_typeck/check/cast.rs:370
+   (~~~~ LINES REMOVED BY ME FOR BREVITY ~~~~)
+  33: rustc_driver::driver::compile_input
+             at /home/user/rust/src/librustc_driver/driver.rs:1010
+             at /home/user/rust/src/librustc_driver/driver.rs:212
+  34: rustc_driver::run_compiler
+             at /home/user/rust/src/librustc_driver/lib.rs:253
+```
+
+## Getting a backtrace for errors
+[getting-a-backtrace-for-errors]: #getting-a-backtrace-for-errors
+
+If you want to get a backtrace to the point where the compiler emits
+an error message, you can pass the `-Z treat-err-as-bug`, which
+will make the compiler panic on the first error it sees.
+
+This can also help when debugging `delay_span_bug` calls - it will make
+the first `delay_span_bug` call panic, which will give you a useful backtrace.
+
+For example:
+```
```

Comment:
should be ` ```text`

Reply:
Its shell commands and comments, so made it bash! Hope its fine!